### PR TITLE
mgr/dashboard: telemetry activate: show ident fields when checked

### PR DESCRIPTION
**Pull request by aaryanporwal**
Fixes: https://tracker.ceph.com/issues/51020
Before (checking the ident checkbox wasn't doing anything):

image

After (checking ident box shows contact info form):

identChecked

Signed-off-by: Aaryan Porwal <aaryanporwal2233@gmail.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by alfonsomthd — CHANGES_REQUESTED** [review]
Can you add a unit test to avoid regressions?

**Comment by avanthakkar** [comment]
@aaryanporwal I think something like this should be enough for the unit testing:
```
 it('should show ident fields when checked ', () => {
      component.showContactInfo = false;
      component.toggleIndent();
      expect(component.showContactInfo).toBe(true);

    });

**Comment by pereman2** [comment]
it isn't working for me in chrome, do yours work guys? 

https://user-images.githubusercontent.com/30913090/122364825-6ac16e80-cf5a-11eb-9008-6df3dcef5394.mp4

**Review by nizamial09 — APPROVED** [review]
Tested and its working! LGTM @aaryanporwal Once the tests are added I think its good to go.

**Comment by aaryanporwal** [comment]
Jenkins retest this please

**Review by pereman2 — APPROVED** [review]
working fine!!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts [hunk 1]
```diff
@@ -80,6 +80,12 @@ describe('TelemetryComponent', () => {
       expect(component).toBeTruthy();
     });
 
+    it('should show ident fields when checked ', () => {
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts:83]: @nizamial09 @aaryanporwal This test is not helpful as it's not checking if the fields are hidden/shown on checking/unchecking, it's testing that the flag variable is updated accordingly, but the only way to guarantee that the fields are shown/hidden is to effectively check the visibility.
This is my suggestion:
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html
index 3aa55ddf9f8..e3c0a6c1739 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html
@@ -127,7 +127,7 @@
                          class="custom-control-input"
                          id="channel_ident"
                          formControlName="channel_ident"
-                         (click)="toggleIndent()">
+                         (click)="toggleIdent()">
                   <label class="custom-control-label"
                          for="channel_ident"></label>
                 </div>
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts
index f60196f701e..ac3a44d215c 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts
@@ -80,10 +80,27 @@ describe('TelemetryComponent', () => {
       expect(component).toBeTruthy();
     });
 
-    it('should show ident fields when checked ', () => {
-      component.showContactInfo = false;
-      component.toggleIndent();
-      expect(component.showContactInfo).toBe(true);
+    it('should show/hide ident fields when checking/unchecking', () => {
+      const getContactField = () =>
+        fixture.debugElement.nativeElement.querySelector('input[id=contact]');
+      const getDescriptionField = () =>
+        fixture.debugElement.nativeElement.querySelector('input[id=description]');
+
+      // Initially hidden.
+      expect(getContactField()).toBeFalsy();
+      expect(getDescriptionField()).toBeFalsy();
+
+      // Show fields.
+      component.toggleIdent();
+      fixture.detectChanges();
+      expect(getContactField()).toBeTruthy();
+      expect(getDescriptionField()).toBeTruthy();
+
+      // Hide fields.
+      component.toggleIdent();
+      fixture.detectChanges();
+      expect(getContactField()).toBeFalsy();
+      expect(getDescriptionField()).toBeFalsy();
     });
 
     it('should set module enability to true correctly', () => {
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
index 777c38053ad..758612dd472 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
@@ -138,7 +138,7 @@ export class TelemetryComponent extends CdForm implements OnInit {
     );
   }
 
-  toggleIndent() {
+  toggleIdent() {
     this.showContactInfo = !this.showContactInfo;
   }
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html [hunk 2]
```diff
@@ -126,12 +126,14 @@
                   <input type="checkbox"
                          class="custom-control-input"
                          id="channel_ident"
-                         formControlName="channel_ident">
+                         formControlName="channel_ident"
+                         (click)="toggleIndent()">
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html:130]:
```suggestion
                         (click)="toggleIdent()">
```

**Review by alfonsomthd — CHANGES_REQUESTED** [review]

**Comment by aaryanporwal** [comment]
ah I accidentally clicked the "update branch" button on github and it did some bad changes, trying to fix it right now

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts (hunk 1 above)
Comment (aaryanporwal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts]: Thanks! fixed in ad5b3f200529fc0bc511ce99eed338afcaef6a62

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html (hunk 2 above)
Comment (aaryanporwal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html]: done!

**Review by avanthakkar — APPROVED** [review]
LGTM!

**Review by alfonsomthd — APPROVED** [review]
LGTM

**Pull request merged**